spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestBaseWithCatalog.java

```diff
@@ -36,16 +42,55 @@
 import org.apache.iceberg.catalog.SupportsNamespaces;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.inmemory.InMemoryCatalog;
+import org.apache.iceberg.rest.RCKUtils;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.iceberg.rest.RESTCatalogServer;
+import org.apache.iceberg.rest.RESTServerExtension;
 import org.apache.iceberg.util.PropertyUtil;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
 import org.junit.jupiter.api.io.TempDir;
 
 @ExtendWith(ParameterizedTestExtension.class)
 public abstract class TestBaseWithCatalog extends TestBase {
-  protected static File warehouse = null;
+  protected static File warehouse;
+
+  static {
+    try {
+      warehouse = File.createTempFile("warehouse", null);
+      assertThat(warehouse.delete()).isTrue();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static final Map<String, String> CONFIG;
+
+  static {
+    try {
+      CONFIG =
+          Map.of(
+              RESTCatalogServer.REST_PORT, String.valueOf(MetaStoreUtils.findFreePort()),
+              CatalogProperties.WAREHOUSE_LOCATION, warehouse.getAbsolutePath(),
+              // In-memory sqlite database by default is private to the connection that created it.
+              // If more than 1 jdbc connection backed by in-memory sqlite is created behind one
+              // JdbcCatalog, then different jdbc connections could provide different views of table
+              // status even belonging to the same catalog. Reference:
+              // https://www.sqlite.org/inmemorydb.html
+              CatalogProperties.CLIENT_POOL_SIZE, "1");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @RegisterExtension
+  private static RESTServerExtension restServerExtension = new RESTServerExtension(CONFIG);
```

Comment:
I believe we should be able to in-line the config here after all the other comments have been applied

Reply:
Done, config in-lined.